gemrb/core/Game.cpp

```diff
@@ -2022,7 +2022,7 @@ void Game::Infravision()
 			allSelectedWithInfravision &= hasInfravision;
 		}
 
-		if ((someoneWithInfravision && tmp) || !allSelectedWithInfravision) {
+		if ((someoneWithInfravision && tmp) || (!tmp && !allSelectedWithInfravision)) {
```

Comment:
Just a logic flaw, switched testing to some different in BG2 that didn't make me notice this.